test/test_client.py

```diff
@@ -168,3 +169,89 @@ def test_client_for_cluster_fallback_to_aad_device_auth(self):
             ['INFO:pykusto:Failed to get Azure CLI token, falling back to AAD device authentication'],
             cm.output
         )
+
+    def test_1_retry(self):
```

Comment:
maybe we can use a default test retries method and call it with different threholds and arguments